File: test/function-params/function.js
 
/*
 * Author       : OBKoro1
 * CreateDate   : 2020-10-19 18:06:18
 * LastEditors  : OBKoro1
 * LastEditTime : 2020-10-21 20:46:39
 * FilePath     : \koro1FileHeader\test\function-params\function.js
 * Description  : 文件描述
 */

/**
 * @description: 一键生成函数注释 自动解析函数参数
 * @param {*} a
 * @param {*} b
 * @param {*} c
 * @param {array} params
 * @return {*}
 */
function test(a, b, c = '参数默认值', ...params) {

}


class {
  /**
   * description: 
   * param {*} a
   * param {*} b
   * param {*} c
   * return {*}
   */  
  constructor (a,b,c){

  }
  test() {

  }
}

let obj = {
  /**
   * @description: 
   * @param {*} c
   * @param {*} d
   * @param {array} a
   * @return {*}
   */  
  test(c, d = 2, ...a){
    console.log('test111')
  },
  a: () => {

  },
  b: function (a,b,v) {

  }
}


/**
 * description: 
 * param {array} c
 * return {*}
 */
let a = (...c) => c

let b = c => c

/**
 * description: 
 * param {*} c
 * param {*} d
 * return {*}
 */
let b1 = ( c = test(), d) => {}

let a1 = ( c) => {
  console.log('啦啦啦')
}

let a2 = ( c, ...d) => {}

File: test/test-run/function-params/function.java
/*
 * Author       : OBKoro1
 * CreateDate   : 2020-08-16 16:06:55
 * LastEditors  : OBKoro1 obkoro1@foxmail.com
 * LastEditTime : 2022-05-15 10:47:35
 * File         : \fileHead\function-params\function.java
 * Description  : 11111
 */
public class TestMax {
   /** 主方法 */
   /**
    * @description: 
    * @param {String[]} args:
    * @param {String} message:
    * @param {int} n:
    * @return {type}
    */
   public static void main (String[]  args, String message, int n) {
      int i = 5;
      int j = 2;
      int k = max(i, j);
      System.out.println( i + " 和 " + j + " 比较，最大值是34323555111133332221：" + k);
   }

   /**
    * @description: 
    * @param {list<int[],map<string,int>>} num1:
    * @param {list2<int[],map<string,int>>} num2:
    * @return {type}
    */
   public static int max(list<    int[]         , map<string, int>> num1, list2<int[],map<string,int>> num2) {
      int result;
      if (num1 > num2)
         result = num1;
      else
         result = num2;
 
      return result; 
   }
      /**
       * @description: 
       * @param {double...} numbers:
       * @param {int} num2:
       * @return {type}
       */
      public static int max2(double... numbers, int num2) {
          
   }

   public String runtimeException(Exception e) {
   }
   
}

// 一个简单的构造函数
class MyClass {
  int x;
 
  // 以下是构造函数
  MyClass(int i ) {
    x = i;
  }
}